From 12e0f25e3dab3cdf8244061a5315c8aa783d6bf7 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 29 Mar 2024 09:19:28 +0100
Subject: [PATCH 0767/1695] plat-stm32mp1: fix untimely HSE over frequency
 warnings

HSE Monitoring uses LPTIM3 with HSI (divided by 1024).
Then we have to calibrate the HSI beforehand, not at the same time.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e188bf680d034de80e81c8d8973fc084d8bef17f plat-stm32mp1: fix untimely HSE over frequency warnings]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I5e7df965150f1a7b0a21894ed0cc599780e677cc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/370131
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/374134
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_calib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_calib.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_calib.c
index a21b577fb..aeb48b4f3 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_calib.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_calib.c
@@ -573,7 +573,7 @@ static TEE_Result init_stm32mp1_calib(void)
 	return TEE_SUCCESS;
 }
 
-driver_init_late(init_stm32mp1_calib);
+service_init(init_stm32mp1_calib);
 
 typedef TEE_Result (*ta_func)(struct ts_session *s, uint32_t param_types,
 			      TEE_Param params[TEE_NUM_PARAMS]);
-- 
2.47.2

